src/Symfony/Component/HttpFoundation/Tests/BinaryFileResponseTest.php

```diff
@@ -36,7 +36,13 @@ public function testConstruction()
 
     public function testConstructWithNonAsciiFilename()
     {
-        new BinaryFileResponse(__DIR__.'/Fixtures/föö.html', 200, array(), true, 'attachment');
+        touch(sys_get_temp_dir().'/fööö.html');
+
+        $response = new BinaryFileResponse(sys_get_temp_dir().'/fööö.html', 200, array(), true, 'attachment');
+
+        $this->assertSame('fööö.html', $response->getFile()->getFilename());
+
+        @unlink(sys_get_temp_dir().'/fööö.html');
```

Comment:
We should remove the file before the assertion. Otherwise it wouldn't be removed when the assertion failed.